xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-api/src/main/java/org/xwiki/model/reference/EntityReference.java

```diff
@@ -42,6 +43,13 @@
  */
 public class EntityReference implements Serializable, Cloneable, Comparable<EntityReference>
```

Comment:
We should probably have a check in `setParameter` making sure that the value associated with PARENT_TYPE_PARAMETER is valid (trying to set one at all should fail when there is a parent, it should be part of the allowed parents, etc.)

Reply:
If we have this kind of check, should we set it at all in the parameters? Wouldn't it be better to have a dedicated field for it?

Reply:
ok

Reply:
I added a private method to perform all checks: https://github.com/xwiki/xwiki-platform/pull/3698/files#diff-48f0d86889b26bb38588347d5b9d5f42cab06612d6346f8551cba2800ed84f17R321